api/helpers/get-api-health.js

```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  friendlyName: 'Get API health',
+
+  description:
+    'Get the health of the API service. This includes database status and service version.',
+
+  fn: async (_, exits) => {
+    try {
+      const games = await Game.count();
```

Comment:
@lbalmaceda we need a different way to query to `alive` status, if there are no games this returns false. Tested locally http://localhost:1337/health
```
{
  "alive": false,
  "version": "3.1.0"
}
```
which sort of defeats the purpose in my mind. Maybe we can query the port of the db or something similar? cc @itsalaidbacklife

Reply:
An easy fix would be counting with the query to throw an error if there's an issue with the DB adapter (or just any error honestly).
![image](https://user-images.githubusercontent.com/3900123/196025714-5c77b779-6f7b-46d4-b5ff-d1c818bd948c.png)

Their docs actually [suggest the offline scenario](https://sailsjs.com/documentation/concepts/models-and-orm/errors#?adapter-errors):
![image](https://user-images.githubusercontent.com/3900123/196026196-af63c1a3-ca06-4996-b97c-52850a661bea.png)


So I could reorder the try/catch into something like this:

```js
try {
  Game.count(); // ignore result
} catch (err) {
  return false;
}
return true;
```
![image](https://user-images.githubusercontent.com/3900123/196029537-a68ba827-e4b8-46db-a0d3-6a2c06a732d3.png)


If you have a scenario I can reproduce to bring the db down and check for the "db disconnected" status I'm happy to test this approach. I've spent more than an hour trying to stub/mock the database model so I can force it to throw an exception to test it but I couldn't make it work. It's like "Game" is not available from the test context.. And I've tried [lifting sails](https://sailsjs.com/documentation/concepts/testing#?lifecycletestjs) as explained there but that threw an "invalid left assignment" error. I'm blocked at this point. Any clue how I can access this instance to stub it? I've pushed this change with a "skip" for that test that fails. 

----- 

Otherwise, I've been reading the sails docs for [Datastores](https://sailsjs.com/documentation/reference/waterline-orm/datastores). These are like generic adapters for databases, which you can customize if you are using a different db engine i.e. mongodb. However, the datastore API doesn't expose any method to access the connection status. You can always access the "raw driver" to check that but then you'd be tied to the API of the specific provider i.e. mongodb client.

I'm thinking another approach could be using the "config/bootstrap.js" file to define a one-off [script that runs during the app startup](https://sailsjs.com/documentation/reference/configuration/sails-config-bootstrap). In there we could seed/provision the db with an entry. Say we create a new model "DbStatus" with no attributes (will have the defaults). Then we can create a new entry for that model (say we check if none existed before), which we could query from this `helper/get-api-health.js` file.  What do you think?

Reply:
Thanks for the deep dive. So far as I can tell, there's no adapter-agnostic way to interact with the connection directly which will prevent us from testing this properly. That said, because it's fundamentally a dev experience issue rather than user experience, I'm comfortable skipping the tests for the failure case, since they will likely be a pain to configure to work across environments because we use postgresql in production but a mocked in-memory database (sails-disk) during local development.

So I say we take your suggestion above regarding the error handling (catching errors and returning false) and test only the positive case since that's what we have access to. If we figure out a consistent way to tap more directly into the db connection in a cross-environment manner, awesome, and if not then testing this can wait until a larger scale rework of models and dependencies. Great work @lbalmaceda!

Reply:
Yeah! This is my take on the issue, too. Thanks for all the information! LGTM.